### PR TITLE
Images gallery

### PR DESCRIPTION
**Pull request by alex-r-bigelow**
Add the image gallery redesign components

### CODE REVIEW

**Comment by alex-r-bigelow** [comment]
Note that the rebase involved a force push; you'll want to update your repos accordingly

**Comment by brianhelba** [comment]
The last rebase was incorrect. I've rebased again and force-pushed.

**Comment by alex-r-bigelow** [comment]
Thanks... though I think it's still incorrect (this rebase is a mess!). I'll fix and force push again.

**Comment by brianhelba** [comment]
@alex-r-bigelow Where'd you get these SVG icons, particularly the more complex ones for histogram and no image?

Note, for simple ones, like the play / seek icons, I'd rather use upstream CSS icons from Fontello, but they're small, so we can change that later.

**Comment by brianhelba** [comment]
@alex-r-bigelow Also, we need to fix the test failures before we can merge. It looks like Javascript style stuff.

**Comment by alex-r-bigelow** [comment]
I drew them (all my icons are custom... IMO, fontello = ick, but I feel that way about bootstrap too). Most of this is stuff that I'm importing raw from Resonant Lab, so if you want to switch to fontello, that's all on you.

I can't seem to figure out the .eslint configuration; any pointers?

**Comment by alex-r-bigelow** [comment]
Okay, I've fixed the rebase once and for all, and force pushed.

**Comment by brianhelba** [comment]
@alex-r-bigelow  I've fixed the style issues here so the tests all pass, without having to include `eslint-disable`.

File: custom/static/css/bootstrap-responsive.css [hunk 1]
```diff
@@ -997,7 +997,7 @@
   }
   .navbar-inverse .nav-collapse .nav > li > a,
   .navbar-inverse .nav-collapse .dropdown-menu a {
-    color: #999999;
+    color: #444444;
```
Comment (brianhelba) [inline custom/static/css/bootstrap-responsive.css]: Please revert this change. This file isn't used by the new external web app; it's part of legacy code.

File: web_external/js/views/body/ImagesView/ImageWall.js [hunk 2]
```diff
@@ -0,0 +1,339 @@
+/*globals girder, jQuery, d3, Image, Backbone, _*/
+
+var LOG_IMAGE_FETCHES = false; // eslint-disable-line no-unused-vars
```
Comment (brianhelba) [inline web_external/js/views/body/ImagesView/ImageWall.js]: Can this var be removed?

File: web_external/js/views/body/ImagesView/ImageWall.js [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+
+var INITIAL_IMAGES_TO_FETCH = 150; // eslint-disable-line no-unused-vars
```
Comment (brianhelba) [inline web_external/js/views/body/ImagesView/ImageWall.js]: Can this var be removed?

File: web_external/js/views/body/ImagesView/ImageWall.js [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+            self.selectImage(d === self.selectedImageId ? null : d);
+        });
+
+        // Construct a position/height lookup dict
+        // (we want to animate everything at the same
+        // time, so don't actually set these values
+        // until we know them)
+        var placementLookup = {};
+        var tallestHeight = 0;
+        var missingImages = 0; // eslint-disable-line no-unused-vars
```
Comment (brianhelba) [inline web_external/js/views/body/ImagesView/ImageWall.js]: Can this var be removed?

File: web_external/js/views/body/ImagesView/shims/recolorImageFilters.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
+/*globals d3*/
+
+function recolorImageFilters(colorList) { // eslint-disable-line no-unused-vars
```
Comment (brianhelba) [inline web_external/js/views/body/ImagesView/shims/recolorImageFilters.js]: Why is this function in a "shims" directory?

File: web_external/stylesheets/body/imagesPage.styl [hunk 6]
```diff
@@ -1,20 +1,110 @@
...
+      border 1px solid transparent
+      border-radius 3px
+      padding 3px
+
+      img
+        width 1em
+        height 1em
+
+        .selected
+          -webkit-filter url(#recolorImageTo00ABFF)
```
Comment (brianhelba) [inline web_external/stylesheets/body/imagesPage.styl]: How does this work? Is there somewhere I can read up on this behavior?

File: web_external/js/views/body/ImagesView/shims/recolorImageFilters.js (hunk 5 above)
Comment (alex-r-bigelow) [inline web_external/js/views/body/ImagesView/shims/recolorImageFilters.js]: This is sort of a general purpose function that I just copied in from Resonant Lab. I'm sure I'll be bringing more soon (e.g. I rolled my own SVG text wrapping logic, etc) that will also populate the shims directory.

In theory, these things are general-purpose tools and probably *should* be separated into their own repositories + installed via NPM... but they're tiny and that route is definitely overkill at the moment.

File: web_external/stylesheets/body/imagesPage.styl (hunk 6 above)
Comment (alex-r-bigelow) [inline web_external/stylesheets/body/imagesPage.styl]: This is exactly what that weird shim does—it makes SVG filters available in CSS.

File: custom/static/css/bootstrap-responsive.css (hunk 1 above)
Comment (alex-r-bigelow) [inline custom/static/css/bootstrap-responsive.css]: Whoops, that's what I get for being rushed (e.g. project-wide find/replace...). Will fix and commit
